### ISSUE TITLE
dotnet build fails during pulumi new

### ISSUE THREAD

**Issue by alastairs** [pulumi/pulumi#3523]
I am creating a Pulumi project within a repo that specifies a .NET Core `global.json` file to pin the SDK version to 2.2.100, and so receive the following error when I run `pulumi new csharp`:

```plain
C:\Program Files\dotnet\sdk\2.2.108\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(137,5): error NETSDK1045: The current .NET SDK does not support targeting .NET Core 3.0.  Either target .NET Core 2.2 or lower, or use a version of the .NET SDK that supports .NET Core 3.0. [F:\code\platform\infra\pulumi\Infra.csproj]
```

It would probably be worth running `dotnet --version` during the scaffolding process to identify the effective SDK version for the project, and generating a global.json targeting .NET Core SDK 3.0 or higher with `dotnet new global.json --sdk-version 3` as part of the scaffolding before running `dotnet build`.

**Comment by mikhailshilkov** [comment]
The error message looks pretty clear, so I guess a user could find their way to do so?

**Comment by alastairs** [comment]
Yes, the user can find their way to doing that in many cases (as I did here). I feel it would be a worthwhile user experience improvement to do this automatically, though, so that `pulumi new csharp` works out of the box.

**Comment by mikhailshilkov** [comment]
I think https://github.com/pulumi/templates/issues/269 is a better issue to track, so I'll close this one as a dupe.
